### PR TITLE
Nerfs tesla engine by making miniballs only act 20% of the time, also makes tesla bolts no longer bounce indefinitely

### PR DESCRIPTION
**Pull request by ChesterTheCheesy**
oh shit, here we go again
# General Documentation
### Intent of your Pull Request
Nerfs the tesla to not create as much ridiculous amounts of power as it does, since each miniball zap has half the charge of tesla itself. By simple math, currently it means tesla makes 1(main tesla) + 5 (10 miniballs) = 6 tesla power. With change, it'll make 1 + 5/5 = 2, theoritically reducing the power output of tesla engines 3 times. It also means, that once the tesla goes loose, instead of 11 explosions each tick, it makes around 3, hopefully being enough for the server to catch up, reducing the lag.

Tesla bolts, now upon hitting a structure, bounce twice, if the shock came from tesla, or once, if it came from miniball. No more infinite loops because whole station is made of conductive metal.

### Why is this change good for the game?
nerfs tesla and makes it lag less during tesloose

# Wiki Documentation
### Briefly describe your PR and the impacts of it, in layman's terms. 
miniballs now act only 20% of time, tesla bolts no longer bounce between objects indefinitely

### What general grouping does this PR fall under? 
Engineering tweak

### If there are any numerical values involved in your PR that will be relevant to a player, please note them here. 
miniballs now act only 20% of time, meaning power output should be 3 times less for tesla

# Changelog
:cl:  
bugfix: tesla shocks no longer bounce between structures indefinitely
tweak: nerfs tesla by making the miniballs only act 20% of the time
/:cl:

### CODE REVIEW

**Comment by ToasterBiome** [comment]
can we have lag generator back in cargo yet

**Comment by ChesterTheCheesy** [comment]
> 
> 
> can we have lag generator back in cargo yet

i'd love it, but since theos is head coder now, it's not gonna happen soon

File: code/modules/power/tesla/energy_ball.dm [hunk 1]
```diff
@@ -73,6 +70,8 @@
 				collide(E)
 
 		for (var/ball in orbiting_balls)
+			if(prob(80))  //tesla nerf/reducing lag, each miniball now has only 20% to trigger the zap
```
Comment (Bibby0110) [inline code/modules/power/tesla/energy_ball.dm:73]: just make it a define :amlooking:

Comment (ChesterTheCheesy) [reply code/modules/power/tesla/energy_ball.dm:73]: how and why would i make one small prob if into "define"?

Comment (ChesterTheCheesy) [reply code/modules/power/tesla/energy_ball.dm:73]: how and why would i change one small RNG check into define

**Comment by JamieD1** [comment]
@SomeguyManperson

**Review by SomeguyManperson — APPROVED** [review]
